addons/web/static/src/js/views/graph/graph_owl_renderer.js

```diff
@@ -0,0 +1,947 @@
+         * @returns {Object} the chart options used for the current mode
+         */
+        _prepareOptions(datasetsCount) {
+            return {
+                maintainAspectRatio: false,
+                scales: this._getScaleOptions(),
+                legend: this._getLegendOptions(datasetsCount),
+                tooltips: this._getTooltipOptions(),
+                elements: this._getElementOptions(),
+            };
+        }
+
+        /**
+         * Returns an object used to style chart elements independently from the datasets.
+         *
+         * @private
+         * @returns {Object}
+         */
+        _getElementOptions() {
+            const elementOptions = {};
+            if (this.props.mode === 'bar') {
+                elementOptions.rectangle = {borderWidth: 1};
+            } else if (this.props.mode === 'line') {
+                elementOptions.line = {
+                    tension: 0,
+                    fill: false,
+                };
+            }
+            return elementOptions;
+        }
+
+        /**
+         * Returns the options used to generate chart tooltips.
+         *
+         * @private
+         * @returns {Object}
+         */
+        _getTooltipOptions() {
+            const tooltipOptions = {
+                // disable Chart.js tooltips
+                enabled: false,
+                custom: this._customTooltip.bind(this),
+            };
+            if (this.props.mode === 'line') {
+                tooltipOptions.mode = 'index';
+                tooltipOptions.intersect = false;
+            }
+            return tooltipOptions;
+        }
+
+        /**
+         * This function creates a custom HTML tooltip.
+         *
+         * @private
+         * @param {Object} tooltipModel see chartjs documentation
+         */
+        _customTooltip(tooltipModel) {
+            if (this.tooltip) {
+                this.tooltip.remove();
+            }
+
+            if (tooltipModel.opacity === 0) {
+                return;
+            }
+            if (tooltipModel.dataPoints.length === 0) {
+                return;
+            }
+
+            const chartArea = this.chart.chartArea;
+            const chartAreaLeft = chartArea.left;
+            const chartAreaRight = chartArea.right;
+            const chartAreaTop = chartArea.top;
+            const rendererTop = this.el.getBoundingClientRect().top;
+
+            const maxTooltipLabelWidth = Math.floor((chartAreaRight - chartAreaLeft) / 1.68) + 'px';
+            const tooltipItems = this._getTooltipItems(tooltipModel);
+
+            const template = document.createElement('template');
+            const tooltipHtml = qweb.render('GraphView.CustomTooltip', {
+                measure: this.props.fields[this.props.measure].string,
+                tooltipItems: tooltipItems,
+                maxWidth: maxTooltipLabelWidth,
+            });
+            template.innerHTML = tooltipHtml;
```

Comment:
why don't you use tooltipHtml directly without defining template?

Reply:
Because I need to do an `appendChild`, but to do that the html returned by `qweb.render` has to become  a `Node`. This is done through creating a "fake" element, and inserting the html as `innerHtml`. Then I can finally append the created inner `Node` to the actual container.

Is there something I'm not getting right?